Sources/Basic/FileSystem.swift

```diff
@@ -372,86 +374,42 @@ private class LocalFileSystem: FileSystem {
-        // Get the mode we need to set.
-        guard let setMode = setmode(mode.cliArgument) else {
-            throw FileSystemError(errno: errno)
-        }
-        defer { setMode.deallocate() }
+        func setMode(url: URL) throws {
+            // Skip if only files should be changed.
+            if options.contains(.onlyFiles) && isDirectory(AbsolutePath(url.path)) {
+                return
+            }
 
-        let recursive = options.contains(.recursive)
-        // If we're in recursive mode, do physical walk otherwise logical.
-        let ftsOptions = recursive ? FTS_PHYSICAL : FTS_LOGICAL
+            let attrs = try FileManager.default.attributesOfItem(atPath: url.path)
 
-        // Get handle to the file hierarchy we want to traverse.
-        let paths = CStringArray([path.pathString])
-        guard let ftsp = fts_open(paths.cArray, ftsOptions, nil) else {
-            throw FileSystemError(errno: errno)
+            // Compute the new mode for this file.
+            let currentMode = attrs[.posixPermissions] as! Int16
+            let newMode = mode.setMode(currentMode)
+            guard newMode != currentMode else { return }
+            try FileManager.default.setAttributes([.posixPermissions : newMode],
+                                                  ofItemAtPath: url.path)
         }
-        defer { fts_close(ftsp) }
-
-        // Start traversing.
-        while let p = fts_read(ftsp) {
-
-            switch Int32(p.pointee.fts_info) {
-
-            // A directory being visited in pre-order.
-            case FTS_D:
-                // If we're not recursing, skip the contents of the directory.
-                if !recursive {
-                    fts_set(ftsp, p, FTS_SKIP)
-                }
-                continue
-
-            // A directory couldn't be read.
-            case FTS_DNR:
-                // FIXME: We should warn here.
-                break
-
-            // There was an error.
-            case FTS_ERR:
-                fallthrough
 
-            // No stat(2) information was available.
-            case FTS_NS:
-                // FIXME: We should warn here.
-                continue
+        let urlPath = URL(fileURLWithPath: path.pathString)
+        try setMode(url: urlPath)
+        guard isDirectory(AbsolutePath(urlPath.path)) else { return }
 
-            // A symbolic link.
-            case FTS_SL:
-                fallthrough
-
-            // A symbolic link with a non-existent target.
-            case FTS_SLNONE:
-                // The only symlinks that end up here are ones that don't point
-                // to anything and ones that we found doing a physical walk.  
-                continue
-
-            default:
-                break
-            }
-
-            // Compute the new mode for this file.
-            let currentMode = mode_t(p.pointee.fts_statp.pointee.st_mode)
+        guard let traverse = FileManager.default.enumerator(
+            at: urlPath,
```

Comment:
The `urlPath` is only really needed here so maybe just convert path to URL directly here.